online-shop/src/app/store/actions/products.actions.ts

```diff
@@ -0,0 +1,16 @@
+import {createAction, props} from "@ngrx/store";
+import {ProductModelDisplay} from "../../types/product-display.model";
+
+export const loadProducts = createAction(
+  '[Product] Load Products'
+)
+
+export const loadProductsSuccess = createAction(
+  '[Product] Load Products Success',
+  props<{ products: ProductModelDisplay[] }>()
+);
+
+export const loadTodosFailure = createAction(
+  '[Product] Load Products Failure',
+  props<{ error: string }>()
```

Comment:
Also this could be a solution for 'any' thing that I mentioned :) If you think you will have only string, it's not necessary to implement another interface